### PR TITLE
Run rubocop on Circle CI and update rubocop rules

### PR DESCRIPTION
**Pull request by mitchellhenke**
Similar to https://github.com/18F/identity-idp/pull/4055

CodeClimate runs a different version and set of rules for Rubocop than we have in our Gemfile and `.rubocop.yml`. This PR moves the rubocop check into CircleCI so running rubocop locally and the checks in CI are the same. Rules from the IDP were brought over to more closely match. I also removed slim since it wasn't being used.

### CODE REVIEW

File: .codeclimate.yml [hunk 1]
```diff
@@ -62,9 +62,6 @@ plugins:
       - XXX
   pep8:
     enabled: true
-  rubocop:
```
Comment (zachmargolis) [inline .codeclimate.yml:65]: maybe we just remove all codeclimate like we did in the IDP?

File: .circleci/config.yml [hunk 2]
```diff
@@ -61,7 +61,7 @@ jobs:
 
             make test
             yarn test
-            bundle exec slim-lint app/views
```
Comment (aduth) [inline .circleci/config.yml:64]: Should we remove `.slim-lint.yml` ?

File: .rubocop.yml [hunk 3]
```diff
@@ -22,14 +20,19 @@ AllCops:
...
+    - 'public/**/*'
+    - 'node_modules/**/*'
   TargetRubyVersion: 2.5
   TargetRailsVersion: 6.1
   UseCache: true
+  DisabledByDefault: true
 
 Rails:
-  Enabled: true
+  Enabled: false
```
Comment (aduth) [inline .rubocop.yml:33]: Curious if I'm following correctly. If we disable rules by default, do we need to have any of these where we explicitly set `Enabled: false` ?

**Review by aduth — APPROVED** [review]
Nice 👍

**Review by zachmargolis — APPROVED** [review]
LGTM! love this

File: .circleci/config.yml (hunk 2 above)
Comment (mitchellhenke) [reply .circleci/config.yml:64]: Yep, thank you

File: .rubocop.yml (hunk 3 above)
Comment (mitchellhenke) [reply .rubocop.yml:33]: We do not, will remove

File: .codeclimate.yml (hunk 1 above)
Comment (mitchellhenke) [reply .codeclimate.yml:65]: I think we still have code climate in the IDP? 
https://github.com/18F/identity-idp/blob/main/.codeclimate.yml

**Comment by zachmargolis** [comment]
LOL @ code coverage decreasing... LMK if you need an admin merge

**Comment by mitchellhenke** [comment]
@zachmargolis code climate will not beat me this time https://github.com/18F/identity-dashboard/pull/417/commits/ff4e4b6813801124a97a49938d0a044aba7789e6

**Pull request merged**